code-challenges.test.js

```diff
@@ -53,4 +138,42 @@ const numbersToAdd3 = []
 // Expected output: []
 
 
+// describe("sumArrayNums", () => {
+//     it("returns an array of the accumulating sum. An empty array should return an empty array.", () => {
+//       expect(sumArrayNums(numbersToAdd1)).toEqual([2, 6, 51, 60])
+//       expect(sumArrayNums(numbersToAdd2)).toEqual([0, 7, -1, 11])
+//       expect(sumArrayNums(numbersToAdd3)).toEqual([])
+//     })
+//   })
+
+// Fail. No reference error. 
+// Pass
+
 // b) Create the function that makes the test pass.
+
+// PseudoCode:
+// Declare a function names sumArrayNums
+// Set parameter: arrayOfNums
+// Delcare a varibale called newArray with [] as its value
+// set up a for loop
+// Delcare a variable called accumulatingSum and set its value to 0
+// Set accumulatingSum += arrayOfNums[i]
+// Use newArray.push and pass accumulatingSum as the arguments
+// return newArray
+
+
+const sumArrayNums = (arrayOfNums) => {
+    let newArray = []
+    let accumulatingSum = 0
+    for(let i = 0; i < arrayOfNums.length; i++) {
+
+        accumulatingSum += arrayOfNums[i]
+        newArray.push(accumulatingSum)
+      }
+      return newArray
+    
+}
+// Pass
+// console.log(sumArrayNums(numbersToAdd1)) // [2, 6, 51, 60]
+// console.log(sumArrayNums(numbersToAdd2)) // [0, 7, -1, 11]
+// console.log(sumArrayNums(numbersToAdd3)) // []
```

Comment:
William! This is awesome!   I am so proud of you!  You took the feedback and now have created beautiful dynamic functions.  They are very clean.  Your tests are on point, also.  You seem to really understand dynamic functions.  To stretch yourself, try doing number 3 using the reduce method, or test for any edge cases you come across. You are doing great!